addons/web/static/src/core/template_inheritance.js

```diff
@@ -0,0 +1,244 @@
+                            XPathResult.ORDERED_NODE_ITERATOR_TYPE
+                        );
+                        const locations = [];
+                        let loc;
+                        while ((loc = result.iterateNext())) {
+                            locations.push(loc);
+                        }
+                        for (const loc of locations) {
+                            loc.firstChild.replaceWith(target.cloneNode(true)); // cannot be done in while
+                        }
+                        if (target.parentElement) {
+                            const nodes = getNodes(target, operation);
+                            target.replaceWith(...nodes);
+                        } else {
+                            let operationContent = null;
+                            let comment = null;
+                            for (const child of operation.childNodes) {
+                                if (child.nodeType === Node.ELEMENT_NODE) {
+                                    operationContent = child;
+                                    break;
+                                }
+                                if (child.nodeType === Node.COMMENT_NODE) {
+                                    comment = child;
+                                }
+                            }
+                            source = operationContent.cloneNode(true);
+                            if (target.hasAttribute("t-name")) {
+                                source.setAttribute("t-name", target.getAttribute("t-name"));
+                            }
+                            if (comment) {
+                                source.prepend(comment);
+                            }
+                        }
+                        break;
+                    }
+                    case "inner":
+                        while (target.firstChild) {
+                            target.removeChild(target.lastChild);
+                        }
+                        target.append(...operation.childNodes);
+                        break;
+                    default:
+                        throw new Error(`Invalid mode attribute: '${mode}'`);
+                }
+                break;
+            }
+            case "attributes": {
+                for (const child of operation.children) {
+                    if (child.tagName !== "attribute") {
+                        continue;
+                    }
+                    const attributeName = child.getAttribute("name");
+                    const firstNode = child.childNodes[0];
+                    let value = firstNode?.nodeType === Node.TEXT_NODE ? firstNode.data : "";
+
+                    if (child.hasAttribute("add") || child.hasAttribute("remove")) {
+                        if (firstNode?.nodeType === Node.TEXT_NODE) {
+                            throw new Error();
+                        }
+                        const separator = child.getAttribute("separator", ",");
```

Comment:
getAttribute doesn't take a second argument as far as I'm aware. Also this default seems to clash with the other default that you have for joining back at the end. Seems like a direct translation from the python which also has this weird quirk (though in python the second param for get is the default value, but here it's ignored)